src/tests/Makefile.am

```diff
@@ -74,9 +74,18 @@ check-container-debian-sid-image: check-container-%-image:
 	echo "RUN apt-get install -y autoconf automake pkg-config intltool libglib2.0-dev \
 	                             xsltproc docbook-xsl docbook-xml iproute2 iptables ipset ebtables \
 	                             nftables libxml2-utils libdbus-1-dev libgirepository1.0-dev \
-	                             python3-dbus python3-gi python3-nftables \
-	                             procps network-manager gir1.2-nm-1.0 time" && \
-	echo "COPY . /tmp/firewalld"; \
+	                             python3-dbus python3-gi python3-nftables sudo \
+	                             procps network-manager gir1.2-nm-1.0 time policykit-1 systemd" && \
+	echo "ENV container docker" && \
+	echo "VOLUME [ \"/sys/fs/cgroup\" ]" && \
+	echo "CMD [\"/usr/sbin/init\"]" && \
+	echo "RUN mkdir -p /run/dbus" && \
+	echo "COPY . /root/firewalld" && \
+	echo "WORKDIR /root/firewalld" && \
+	echo "RUN ./autogen.sh" && \
+	echo "RUN ./configure --sysconfdir=/etc PYTHON=\"/usr/bin/python3\" am_cv_python_pythondir=/usr/lib/python3/dist-packages" && \
+	echo "RUN make" && \
+	echo "RUN make install" ; \
```

Comment:
these containerfiles are constructed by shell commands, inside the Makefile.am.

It would have been better, if this were a separate script (and/or a separate containerfile) that is commited to git. That makes the functionality re-usable without make. It also makes the content easier to review, as there are not multiple laters of escaping. The script of course can still be called from make.

Reply:
I don't think there is a use case outside of the build/make/test? I did think about commonizing these dockerfiles, but they're actually different enough (packages, configure, alternatives) that it didn't seem useful. It would be more work to mix-in the differences.